src/app/games/diam/diam.component.ts

```diff
@@ -0,0 +1,320 @@
+                const fgClasses: string[] = [];
+                if (this.selected != null &&
+                    this.selected.type === 'pieceFromReserve' &&
+                    this.selected.piece === piece &&
+                    y === remaining-1) {
+                    // highlight the top piece of the remaining pieces stack if it is selected
+                    fgClasses.push('highlighted');
+                }
+                if (y === remaining-1 && piece.owner === currentPlayer) {
+                    // Only let the top piece be clickable
+                    fgClasses.push('clickable-hover');
+                }
+                this.viewInfo.remainingPieces.push({
+                    bgClasses: ['player' + piece.owner.value + (piece.otherPieceType ? '-alternate' : '')],
+                    fgClasses,
+                    y,
+                    animate: false,
+                    drawPositionEnd: this.getDrawPositionRemainingPiece(piece, y),
+                    drawPositionStart: MGPOptional.empty(),
+                    actualPiece: piece,
+                });
+            }
+        }
+    }
+    private getDrawPositionRemainingPiece(piece: DiamPiece, y: number): Coord {
+        let x: number;
+        if (piece.owner === Player.ZERO) {
+            x = -100;
+        } else {
+            x = 600;
+        }
+        let initialY: number;
+        if (piece.otherPieceType) {
+            initialY = 100;
+        } else {
+            initialY = 450;
+        }
+        return new Coord(x, initialY).getNext(new Vector(0, -y*DiamComponent.PIECE_HEIGHT));
+    }
+    private getCaseClasses(x: number): string[] {
+        const lastMove: DiamMove | null = this.rules.node.move;
+        if (lastMove != null) {
+            if (lastMove.isDrop()) {
+                if (lastMove.getTarget() === x) {
+                    return ['moved'];
+                }
+            } else {
+                if (lastMove.getTarget() === x || lastMove.start.x === x) {
+                    return ['moved'];
+                }
+            }
+        }
+        return [];
+    }
+    private getPieces(x: number): PieceInfo[] {
+        const highestAlignment: MGPOptional<Coord> = this.rules.findHighestAlignment(this.rules.node.gameState);
+        const infos: PieceInfo[] = [];
+        for (let y: number = DiamState.HEIGHT-1; y >= 0; y--) {
+            const piece: DiamPiece = this.rules.node.gameState.getPieceAtXY(x, y);
+            const coord: Coord = new Coord(x, y);
+            if (piece !== DiamPiece.EMPTY) {
+                const fgClasses: string[] = [];
+                let animate: boolean = false;
+                let drawPositionStart: MGPOptional<Coord> = MGPOptional.empty();
+                if (this.selected != null &&
+                    this.selected.type === 'pieceFromBoard' &&
+                    this.selected.position.x === x &&
+                    this.selected.position.y === y) {
```

Comment:
this.isHighlightable(this.selected) :p ?

Reply:
fait